Modules/SubModule_Import/xASL_bids_BIDSifyCheckTimeEncoded.m

```diff
@@ -0,0 +1,90 @@
+function [jsonIn,jsonOut,bTimeEncoded,bTimeEncodedFME] = xASL_bids_BIDSifyCheckTimeEncoded(jsonIn,jsonOut)
+%xASL_bids_BIDSifyCheckTimeEncoded Check for time encoded sequence
+%
+% FORMAT: [jsonIn,jsonOut] = xASL_bids_BIDSifyCheckTimeEncoded(jsonIn,jsonOut)
+%
+% INPUT:
+%   jsonIn    - JSON with the input fields - from DICOMs (REQUIRED)
+%   jsonOut   - ordered and checked JSON structure
+%
+% OUTPUT: 
+%   jsonIn          - JSON with the input fields - from DICOMs
+%   jsonOut         - JSON output structure
+%   bTimeEncoded    - Boolean describing if the current sequence is a time encoded sequence
+%   bTimeEncodedFME - Boolean describing if the current sequence is a specific FME time encoded sequence
+%
+% -----------------------------------------------------------------------------------------------------------------------------------------------------
+% DESCRIPTION: Check for time encoded sequence.
+%
+% EXAMPLE:     [jsonIn,jsonOut,bTimeEncoded,bTimeEncodedFME] = xASL_bids_BIDSifyCheckTimeEncoded(jsonIn,jsonOut);
+%
+% __________________________________
+% Copyright 2015-2021 ExploreASL
+
+    % Check for time encoded sequences
+    if isfield(jsonOut,'TimeEncodedMatrixSize') && ~isempty(jsonOut.TimeEncodedMatrixSize) || ... % Should be 4, 8 or 12
+        isfield(jsonOut,'TimeEncodedMatrixType') % Natural or walsh
+        bTimeEncoded = true; 
+    else 
+        bTimeEncoded = false; 
+    end
+
+
+    % Check for specific time encoded sequence of FME (Fraunhofer Mevis)
+    if isfield(jsonIn,'SeriesDescription')
+        bTimeEncodedFME = ~isempty(regexp(jsonIn.SeriesDescription,'(Encoded_Images_Had)\d\d(_)\d\d(_TIs_)\d\d(_TEs)', 'once'));
+    else
+        bTimeEncodedFME = false;
+    end
+
+    if bTimeEncodedFME
+        bTimeEncoded = true;    
+    end
+
+    if bTimeEncoded
+        if isfield(jsonOut,'EchoTime') && isfield(jsonOut,'PostLabelingDelay')
+            % From the import, the length of EchoTime should correspond to the number of volumes
+            if length(jsonOut.EchoTime)~=length(jsonOut.PostLabelingDelay)
+                % So here, we first make sure that each PLD is repeated for the whole block of echo-times
+                jsonOut.TimeEncodedNumberTE = length(uniquetol(jsonOut.EchoTime,0.001)); % Obtain the number of echo times
```

Comment:
TimeEncodedNumberTE is a parameter that is not used outside of this function. Also - it does not make sense to make it part of the output JSON, because in BIDS, we simply can obtain it from the vector of TE. 

I would thus take it out of jsonOut and keep locally only - this won't affect the code in any way.
jsonOut.TimeEncodedNumberTE -> numberTEs

Reply:
Related to that, you have to specify numberTEs = 1; at line 43
And also, at line 76, change "isfield(jsonOut,'TimeEncodedNumberTE')" to "numberTEs>1"

Reply:
Fixed e916b6d 👍